mesonbuild/compilers/compilers.py

```diff
@@ -79,23 +75,8 @@
 clink_suffixes += ('h', 'll', 's')
 all_suffixes = set(itertools.chain(*lang_suffixes.values(), clink_suffixes))
 
-# Languages that should use LDFLAGS arguments when linking.
-languages_using_ldflags = {'objcpp', 'cpp', 'objc', 'c', 'fortran', 'd', 'cuda'}
-# Languages that should use CPPFLAGS arguments when linking.
-languages_using_cppflags = {'c', 'cpp', 'objc', 'objcpp'}
 soregex = re.compile(r'.*\.so(\.[0-9]+)?(\.[0-9]+)?(\.[0-9]+)?$')
 
-# Environment variables that each lang uses.
-cflags_mapping = {'c': 'CFLAGS',
-                  'cpp': 'CXXFLAGS',
-                  'cuda': 'CUFLAGS',
-                  'objc': 'OBJCFLAGS',
-                  'objcpp': 'OBJCXXFLAGS',
-                  'fortran': 'FFLAGS',
-                  'd': 'DFLAGS',
-                  'vala': 'VALAFLAGS',
-                  'rust': 'RUSTFLAGS'}
```

Comment:
I'm not sure if those lists are better in compilers or environment. I have a slight preference in keeping this here (capitalize the dict name is fine), but it can be seen both ways I guess.

Reply:
@xclasses The idea is we *really* don't want arbitrary code accessing environment variables. All config, by it from CLI, env vars, or configuration files, should be gathered in one place, and the rest of the code should not have to care about where the raw options have come from.

Having completely different code paths for the different ways to configure has been a huge source of bugs in the past, and I believe is one of the reasons environment variables have gotten such a bad rep around here. (Though don't get me wrong, they are still iffy in general.)

Reply:
I'm all for having the code in a single place. The point here is the mapping language<>env is more a compiler matter. The code actually taking the value is an Envorinment.py matter. But I guess that can be argued both ways, I don't have strong opinion against this change.

Reply:
I like moving it just to prevent spaghetti and the X imports Y imports X problem. I can make this module "private" if that makes it better?